training/training/routes/datasets/default/columns.py

```diff
@@ -1,18 +1,25 @@
 from django.http import HttpRequest
 from ninja import Router, Schema
+from ninja.security import HttpBearer
 from training.core.dataset import SklearnDatasetCreator
 from training.routes.datasets.default.schemas import DefaultDatasetResponse
 from training.routes.schemas import NotFoundError
+from core.authenticator import FirebaseAuth
 
 router = Router()
 
 
 @router.get(
     "{name}/columns",
     response={200: DefaultDatasetResponse, 404: NotFoundError},
+    auth=FirebaseAuth(),
 )
 def defaultDatasets(request: HttpRequest, name: str):
     if not name in SklearnDatasetCreator.DEFAULT_DATASETS:
         return 404, {"message": "Dataset not found"}
     dataset = SklearnDatasetCreator.getDefaultDataset(name)
-    return 200, {"data": dataset.columns.tolist(), "message": "Success"}
+    return 200, {
+        "data": dataset.columns.tolist(),
+        "message": "Success",
+        "token": request.auth,
```

Comment:
Do this to satisfy type hints for `request.auth`: [https://github.com/vitalik/django-ninja/issues/508](https://github.com/vitalik/django-ninja/issues/508)

Reply:
This would involve changing the first argument of the function header to Request instead of HttpRequest